cdc/server.go

```diff
@@ -356,3 +348,22 @@ func findBestDataDir(candidates []string) (result string, ok bool) {
 
 	return result, ok
 }
+
+func checkAndWaitClusterVersion(
+	ctx context.Context, pdClient pd.Client, pdEndpoints []string, security *config.SecurityConfig,
+) {
+	const backoffOneSecond = 1000
+	retry.Do(ctx, func() error {
+		var err error
+		for _, pdEndpoint := range pdEndpoints {
+			err = version.CheckClusterVersion(ctx, pdClient, pdEndpoint, security)
+			if err == nil {
+				break
+			}
```

Comment:
Two scenarios in this block
- More than 3 PDs in upstream, one of them is down, TiCDC can start up, which is the HA feature, in this scenario TiCDC only check one PD.
- Upstream upgrade, should TiCDC waits all PDs upgrade to the required version, then TiCDC should check all PDs.